crates/bevy_sprite/src/mesh2d/mesh.rs

```diff
@@ -255,16 +255,6 @@ struct BatchMeta2d {
     dynamic_offset: Option<NonMaxU32>,
 }
 
-impl BatchMeta<BatchMeta2d> for BatchMeta2d {
-    fn matches(&self, other: &BatchMeta2d) -> bool {
-        self.pipeline_id == other.pipeline_id
```

Comment:
There is one aspect I've been considering - the order of comparison of members. Maybe it's much faster to compare the thing that is most likely to break a batch first, rather than the possibly least likely thing (the pipeline). Does a derived `PartialEq` compare in struct member order?

Reply:
yes it does seem to. it uses the source AST tokens so it would make sense that it's in declared order.